tycho-core/src/main/java/org/eclipse/tycho/p2/publisher/AbstractMetadataGenerator.java

```diff
@@ -110,7 +110,7 @@ private void createRequirementFromPlatformURL(ArrayList<IRequirement> result, St
         Matcher m = TychoConstants.PLATFORM_URL_PATTERN.matcher(url);
         if (m.matches())
             result.add(MetadataFactory.createRequirement(IInstallableUnit.NAMESPACE_IU_ID, m.group(2),
-                    VersionRange.emptyRange, null, false, false));
+                    VersionRange.emptyRange, null, true, false));
```

Comment:
```suggestion
                    VersionRange.emptyRange, null, true, true));
```
If we make it optional it should be at least greedy.

Reply:
I considered this as well, but according to the doc the second flag doesn't handle greediness but multiplicity:

https://github.com/eclipse-equinox/p2/blob/5b57f58f9ad96c332dcf99218b24dbc5a5cfb705/bundles/org.eclipse.equinox.p2.metadata/src/org/eclipse/equinox/p2/metadata/MetadataFactory.java#L535-L537

Reply:
You need then this method: https://github.com/eclipse-equinox/p2/blob/5b57f58f9ad96c332dcf99218b24dbc5a5cfb705/bundles/org.eclipse.equinox.p2.metadata/src/org/eclipse/equinox/p2/metadata/MetadataFactory.java#L565

Reply:
In fact the currently used method also makes the created requirement greedy:
https://github.com/eclipse-equinox/p2/blob/5b57f58f9ad96c332dcf99218b24dbc5a5cfb705/bundles/org.eclipse.equinox.p2.metadata/src/org/eclipse/equinox/p2/metadata/MetadataFactory.java#L543C39-L543C43

So this is already as desired.